### ISSUE TITLE
Highlight all occurences across all open (visible) files

### ISSUE THREAD

**Issue by jon-walstedt** [zed-industries/zed#21549]
### Check for existing issues

- [X] Completed

### Describe the feature

I am a (neo)vim user that is exploring Zed and I am really happy with it so far, really good work! One thing i am missing though is to highlight/search for a string under the cursor using `*` across all open (visible) files. As it is now it highlights all occurrences but only in the current file. I usually have multiple split views open and would want to have the highlight present in all of them.

It would also be nice if it was possible to move the search bar to the bottom or even if it was possible to opt out of the ui for it.

In the screenshot below i would expect ctrl to be highlighted in both splits

![Screenshot 2024-12-04 at 13 27 06](https://github.com/user-attachments/assets/a214db2c-d641-444d-b66d-9395fd44569f)

### Environment

Zed: v0.163.3 (Zed)
OS: macOS 15.1.1
Memory: 18 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="1230" alt="Screenshot 2024-12-04 at 21 12 06" src="https://github.com/user-attachments/assets/b43dae38-a5a4-481d-b714-77ab9783f5e1">

**Comment by notpeter** [comment]
Interesting.  I can totally understand how this would be useful. Likely this should be a non-default behavior as VSCode, JetBrains, Xcode, SublimeText (and Zed currently) all treat window splits as distinct buffers that can have their own independent selections.

**Comment by jonwalstedt** [comment]
Yeah, having it as a non-default behaviour or part of the vim mode would be fine. My main usage of this (which I use a lot in vim) is to highlight a word under the cursor, cycle through all instances within the current buffer (using `n` and `N` for reverse search)  then jump to another buffer in another split and just keep cycling through all instances there without having to redo the search.

**Comment by KieranP** [comment]
Just quickly chiming in to say that VSCode's default is to highlight all occurrences in the current file, but there is a setting, `editor.occurrencesHighlight` which allows using `multiFile`, which causes it to highlight the same keyword across all visible files. This is something I use quite a bit and have had to revert to using VSCode. Once/if this is implemented, there would be nothing missing in Zed for me to use it daily.

**Comment by Xenomorph439** [comment]
I want to be able to switch this setting off 
I don't want the unnecessary highlighting of text under my cursor

**Comment by Kalmaegi** [comment]
If possible, I’d like to try implementing this feature. We could make it configurable and keep it disabled by default.

My simple idea is that when this option is enabled, the current search could synchronize across all visible panes, allowing users to seamlessly continue searching (e.g., jumping to the next match) even after switching to a different pane.

But I’m not sure whether we should actually add a new variant to the `SearchOptions` struct for this. I think it should be a separate Vim setting, checked at a higher level before triggering the search.

**Comment by Kalmaegi** [comment]
I was wondering whether it might be better to keep this feature specific to Vim mode, or make it available as a global setting. It would also change how we implement it, so maybe keeping it open instead of Vim-mode-only would make more sense.

**Comment by KieranP** [comment]
I don't use Vim mode personally, so would prefer it not be tied to that. The ability to double click to highlight a variable, and see all occurrences of it between multiple visible panes would be very helpful. Now that sticky header got added in todays pre releases, this is the last thing I've been missing from VSCode.
